### PR TITLE
Rerun snv-frequencies with updated PR from OPC repo

### PR DESCRIPTION
**Pull request by kelseykeith**
#### What scientific question is your analysis addressing?

Needed to rerun the `snv-frequencies` module in order to re-generate the results tables with the updated data and updated PedcBio case IDs after the OpenPedCan v11 data release and its push to PedcBio. Additionally, it was decided at this time that CHOP P30/DGD panel and TARGET panel data should be excluded from "All cohorts."

#### What was your approach?

I had to make several modifications to `01-snv-frequencies.R`

First, I updated the `get_cs_cg_tbl()` function to allow specifying the cohorts that should be included in the all cohorts calculation so that panel data would be excluded.

Second, previously, the PedcBio cancer study ID, ex: "ped\_opentargets\_2021" or "opedpedcan\_v11" was hard-coded into several functions. I modified these functions so that the study ID has to be specified to make re-running this module easier in the future. The modified functions are:

- `get_pcb_pot_csi()`
- `get_pcb_pot_plot_url()`
- `add_cg_ch_pedcbio_pedot_plot_urls()`
- `get_cg_ch_var_level_mut_freq_tbl()`

Additionally, for this fix, the PedcBio cancer study ID is now specified as a variable in the `Read data` section of `01-snv-frequencies.R`, ex: `pcb_cancer_study <- 'openpedcan_v11'` for this PR.

Third, the names of the panel cohorts don't match between what we'd like to display on the MTP site and what they are in PedcBio Portal. To match them up, I added a simple two column table, `pedcbio_cohorts`, with one column for MTP cohort names and one column for PedcBio cohort names and modified the functions below to use the appropriate cohort names when making the PedcBio URLs.

- `add_cg_ch_pedcbio_pedot_plot_urls()`
- `get_cg_ch_var_level_mut_freq_tbl()`

Finally, for [Ticket #293](https://github.com/PediatricOpenTargets/ticket-tracker/issues/293), the histology and independent specimen assertion could not be accomplished as written in the ticket. The code given there checks that all patients present in both the histology and SNV MAF files are present in the independent-specimen lists. However, there are a few intended exceptions where that this is not true because those patients have diseases or missing information that excluded them from independent specimens. These patients are not included in the results and are filtered out because they are not in the independent specimens table.

Instead, I added a check to make sure that the histology and independent sample lists match after they are combined, similar to the existing check between the histology and the MAF files.

#### What GitHub issue does your pull request address?

1.  Updated analysis: assert that histology file is compatible with independent sample lists in snv-frequencies module [Ticket #293](https://github.com/PediatricOpenTargets/ticket-tracker/issues/293)
2. Updated analysis: update snv-frequencies for v11 data release [Ticket #361](https://github.com/PediatricOpenTargets/ticket-tracker/issues/361)
3.  Updated analysis: rerun snv-frequencies with updated v11 pedcbio case IDs [Ticket #386](https://github.com/PediatricOpenTargets/ticket-tracker/issues/386)

#### Which areas should receive a particularly close look?

- Check that updated functions behave as expected since they were modified.
- The codes executes correctly and no assertions fail.
- Check that new DGD samples read in correctly and are present in the results.
- Check that cohorts are named appropriately; panel data should be either "CHOP P30 panel" or "TARGET panel"
- Check outputs for consistency, particularly the PedcBio oncoprint and mutation plot URLs. Suggest checking a couple random samples at different mutation frequencies and for different cohorts to make sure the URLs are correct, especially for the new DGD samples and the TARGET panel samples.

#### Is there anything that you want to discuss further?

Updated PedcBio case set IDs are included in the PR, `input/ped_opentargets_v11_pedcbio_case_set_ids.json`, so you should not have to fetch them yourself, but instructions on how to fetch them from PedcBio Portal are included in the module [README](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/snv-frequencies#add-annotations).

Also, for the future, the results are getting large so we may need to transistion to Git Large File Storage (LFS) or store them in AWS buckets going forward.

There are other PRs for these tickets, but @jharenza asked for it to be re-submitted as a branch off the OPC repo so GitHub Actions will function correctly.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

TSV and JSON tables

#### What is your summary of the results?

- `input/ped_opentargets_v11_pedcbio_case_set_ids.json`
- `gene-level-snv-consensus-annotated-mut-freq.jsonl.gz`
- `gene-level-snv-consensus-annotated-mut-freq.tsv.gz`
- `variant-level-snv-consensus-annotated-mut-freq.jsonl.gz`
- `variant-level-snv-consensus-annotated-mut-freq.tsv.gz`

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
Thank you for resubmitting @kelseykeith. I re-requested reviewers over here now.

**Review by ewafula — CHANGES_REQUESTED** [review]
@kelseykeith, thank you for performing extensive updates on this module. Code updates for all related issues look ok except for the inclusion of panel data.  You need to change the logic to exclude the TARGET and  DGD panels from only `all cohorts frequencies calculations` and not in the `All cohorts Datatset type` in the MPT frequencies tables. The approach should be to filter out panel samples from only `all cohorts independent specimen files` and not `each cohort independent specimen files,` similar to the @jharenza's update of the fusions module. I know this is confusing! I, too, was confused until @jharenza  clarified. 

Your SNV results have only the 3 cancer groups in  `All Cohorts` Dataset type column after including the panel data
```
> snv_freq %>% select(Dataset, Disease) %>% distinct() %>% filter(Dataset == "All Cohorts")
# A tibble: 3 x 2
  Dataset     Disease             
  <chr>       <chr>               
1 All Cohorts Ganglioneuroblastoma
2 All Cohorts Ganglioneuroma      
3 All Cohorts Neuroblastoma 
```
Fusion  results have 39 cancer groups in  `All Cohorts` Dataset type column after including the panel data
```
> fusion_freq %>% select(Dataset, Disease) %>% distinct() %>% filter(Dataset == "All Cohorts")
# A tibble: 39 x 2
   Dataset     Disease                               
   <chr>       <chr>                                 
 1 All Cohorts Acute Lymphoblastic Leukemia          
 2 All Cohorts Acute Myeloid Leukemia                
 3 All Cohorts Adamantinomatous Craniopharyngioma    
 4 All Cohorts Atypical Teratoid Rhabdoid Tumor      
 5 All Cohorts Chordoma                              
 6 All Cohorts Choroid plexus carcinoma              
 7 All Cohorts CNS Embryonal tumor                   
 8 All Cohorts Craniopharyngioma                     
 9 All Cohorts Diffuse intrinsic pontine glioma      
10 All Cohorts Dysembryoplastic neuroepithelial tumor
# ... with 29 more rows
> 
```
You also need to either set `guess_max ` in the read_tsv() function when reading the large maf file or  `col_select` only the columns used the script and set datatypes to avoid the warning causing the GA to fail 
```
Read data...
Warning: 28 parsing failures.
    row col               expected  actual                                                file
 464671  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
1869464  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
1892736  AF no trailing characters ,0.1985 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2091025  AF no trailing characters ,0.0640 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
4407213  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
....... ... ...................... ....... ...................................................
See problems(...) for more details.
```
```
Read data...
Warning: 12080 parsing failures.
 row     col           expected actual                                                file
2443 SOMATIC 1/0/T/F/TRUE/FALSE  0,1,1 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2443 PHENO   1/0/T/F/TRUE/FALSE  0,1,1 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2[6](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/runs/7878608083?check_suite_focus=true#step:32:7)98 SOMATIC 1/0/T/F/TRUE/FALSE  0,1   '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2698 PHENO   1/0/T/F/TRUE/FALSE  0,1   '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2[7](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/runs/7878608083?check_suite_focus=true#step:32:8)[13](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/runs/7878608083?check_suite_focus=true#step:32:14) SOMATIC 1/0/T/F/TRUE/FALSE  0,0,1 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
.... ....... .................. ...... ...................................................
See problems(...) for more details.

```
Since GA seems to be failing on warnings such as those shown above. It is likely that when results are not available in the module, the following warning in the bash script will cause the GA to fail 
`rm: cannot remove 'results/*.gz': No such file or directory`

Use this `rm -f results/*.gz`  to remove the gzipped files instead of `rm results/*.gz || true` to suppress the Unix warning message above. 

Cc @jharenza

**Comment by kelseykeith** [comment]
Thank you for the review @ewafula! I will take a look again at the `All cohorts` frequencies calculation, clearly there should be more cancer types with `All cohorts`.

Work on fixing the GA error is in progress!

The ` || true` in `rm results/*.gz || true` allows the bash script to proceed even if there are no results available in the the results folder. I will add the `-f` just in case though.

**Review by adilahiri — COMMENTED** [review]
@kelseykeith thank you for your work on the module. 
I did run the module on EC2 instance got the same warnings as @ewafula has already pointed out

```
Read data...
Warning: 28 parsing failures.
    row col               expected  actual                                                file
 464671  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
1869464  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
1892736  AF no trailing characters ,0.1985 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
2091025  AF no trailing characters ,0.0640 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
4407213  AF no trailing characters ,0.0022 '../../data/snv-consensus-plus-hotspots.maf.tsv.gz'
....... ... ...................... ....... ...................................................
See problems(...) for more details.
```
I am trying to run this through Rstudio on EC2 to get a better understanding of the code.

**Comment by kelseykeith** [comment]
After the last few commits to resolve the GitHub Action problems all issues should be addressed: 

#### GitHub Actions / Warnings
The GitHub Actions error is resolved; it was caused by a parsing issue that I assume is only present in the GitHub Actions test files as I could not recreate it otherwise. The warnings are fine as they don't interfere with either GitHub Actions or successful completion of the module. I did add `rm -f` to the bash execution script just to be safe however.

#### Number of Cancer Groups with "All Cohorts"
Upon re-review of my code, I believe the number of cancer groups with "All cohorts" is correct. If you check the `histologies.tsv` file for WGS/WXS samples with biospecimens present in the MAF file, there are only 3 cancer groups with samples from multiple cohorts. (`n` is the number of samples for that `cancer_group` in that `cohort` and `nn` is the number of `cohort`s in that `cancer_group`) 
```
maf_df %>%
  distinct(Kids_First_Biospecimen_ID) %>%
  left_join(htl_df) %>%
  count(cancer_group, cohort) %>%
  arrange(cancer_group, cohort) %>%
  filter(!str_detect(cohort, 'panel')) %>%
  add_count(cancer_group) %>%
  filter(nn != 1)
  
# A tibble: 8 × 8
cancer_group  cohort  n  nn
<chr>  <chr>  <int>  <int>
Ganglioneuroblastoma  GMKF  36  3 
Ganglioneuroblastoma  PBTA  3  3
Ganglioneuroblastoma  TARGET  34  3
Ganglioneuroma  GMKF  3  2  
Ganglioneuroma  PBTA  1  2  
Neuroblastoma  GMKF  297  3 
Neuroblastoma  PBTA  3  3  
Neuroblastoma  TARGET  188  3
  ```
Additionally, when checking the old results in the `dev` branch, it showed only one cancer group with an "All cohorts" dataset, so the low number of "All Cohorts" datasets is consistent with previous results.
```
gene_res %>% filter(Dataset == 'All Cohorts') %>% distinct(Disease)
# A tibble: 1 × 1
  Disease      
  <chr>        
1 Neuroblastoma
```

File: analyses/snv-frequencies/01-snv-frequencies.R [hunk 1]
```diff
@@ -629,29 +665,63 @@ htl_df <- read_tsv('../../data/histologies.tsv', guess_max = 100000,
...
+                            TRUE ~ cohort))
+
+# The cohort names we want to display on the MTP website do not match the 
+# PedcBio cohort names, so need a 2-column table matching them up with 2 columns:
+# - cohort: the cohort we want displayed on MTP
+# - pedcbio_cohort: the name of that cohort on PedcBio
+htl_df %>%
+  distinct(cohort) %>%
+  mutate(pedcbio_cohort = case_when(cohort == 'TARGET panel' ~ 'TARGET',
+                                    cohort == 'CHOP P30 panel' ~ 'DGD',
```
Comment (jharenza) [inline analyses/snv-frequencies/01-snv-frequencies.R]:
```suggestion
  mutate(cohort = case_when(cohort == 'TARGET' & 
                              experimental_strategy == 'Targeted Sequencing' ~
                              'TARGET Panel',
                            cohort == 'DGD' & 
                              experimental_strategy == 'Targeted Sequencing' ~ 
                              'CHOP P30 Panel',
                            TRUE ~ cohort))

# The cohort names we want to display on the MTP website do not match the 
# PedcBio cohort names, so need a 2-column table matching them up with 2 columns:
# - cohort: the cohort we want displayed on MTP
# - pedcbio_cohort: the name of that cohort on PedcBio
htl_df %>%
  distinct(cohort) %>%
  mutate(pedcbio_cohort = case_when(cohort == 'TARGET Panel' ~ 'TARGET',
                                    cohort == 'CHOP P30 Panel' ~ 'DGD',
```

Hi @kelseykeith can you just update this to be consistent with fusion-freq/ capitalization style. Thanks!

**Comment by kelseykeith** [comment]
Re-running now with updated panel names!

**Comment by jharenza** [comment]
Thanks, sorry about that! I know I wrote it that way first.

**Comment by kelseykeith** [comment]
No problem! Updated tables passed the check so we should be good to go now

**Review by jharenza — APPROVED** [review]
thanks for your work on this @kelseykeith !

**Review by ewafula — APPROVED** [review]

**Review by adilahiri — COMMENTED** [review]
Thank you for the updates @kelseykeith and detailed review @ewafula 

I am going over the module. Module ran successfully on EC2 (OpenPedCan instance with m6i.2xlarge (4 vCPU and 32Gb memory)). The run time was close to five 5 minutes. Smaller machines were taking more than a few hours to run this module. 

Got three cancer groups within All Cohorts. 
```
gene_level_mut_freq_tbl%>% select(Dataset, Disease) %>% distinct() %>% filter(Dataset == "All Cohorts")
# A tibble: 3 x 2
  Dataset     Disease             
  <chr>       <chr>               
1 All Cohorts Ganglioneuroblastoma
2 All Cohorts Ganglioneuroma      
3 All Cohorts Neuroblastoma  
```
```
> var_level_mut_freq_tbl%>% select(Dataset, Disease) %>% distinct() %>% filter(Dataset == "All Cohorts")
# A tibble: 3 x 2
  Dataset     Disease             
  <chr>       <chr>               
1 All Cohorts Ganglioneuroblastoma
2 All Cohorts Ganglioneuroma      
3 All Cohorts Neuroblastoma  
```

**Pull request merged**